src/chart/pie/labelLayout.ts

```diff
@@ -228,6 +255,91 @@ function avoidOverlap(
     }
 }
 
+/**
+ * Set max width of each label, and then wrap each label to the max width.
+ *
+ * @param layout label layout
+ * @param availableWidth max width for the label to display
+ * @param forceRecalculate recaculate the text layout even if the current width
+ * is smaller than `availableWidth`. This is useful when the text was previously
+ * wrapped by calling `layoutText` but now `availableWidth` changed, in which
+ * case, previous wrapping should be redo. `forceRecalculate` is ignored if
+ * `labelStyleWidth` is set.
+ */
+function layoutText(
+    layout: LabelLayout,
+    availableWidth: number,
+    forceRecalculate: boolean = false
+) {
+    const label = layout.label;
+    const style = label.style;
+    const textRect = layout.rect;
+    const bgColor = style.backgroundColor;
+    const padding = style.padding as number[];
+    const paddingH = padding ? padding[1] + padding[3] : 0;
+    const overflow = style.overflow;
+
+    // textRect.width already contains paddingH if bgColor is set
+    const oldOuterWidth = textRect.width + (bgColor ? 0 : paddingH);
+    if (availableWidth < oldOuterWidth
+        || forceRecalculate && layout.labelStyleWidth == null
```

Comment:
I think it's not necessary to enter this whole method if `labelStyleWidth` is not null